eth_utils/abi.py

```diff
@@ -112,7 +112,7 @@ def _get_tuple_type_str_and_dims(s: str) -> Optional[Tuple[str, Optional[str]]]:
     return None
 
 
-def collapse_if_tuple(abi: Union[ABIComponent, str]) -> str:
+def collapse_if_tuple(abi: Union[ABIComponent, Dict[str, Any], str]) -> str:
```

Comment:
Opening this back up to include the original `Dict[str, Any]` to include cases where the ABI element doesn't have a `name`. The [spec](https://docs.soliditylang.org/en/v0.8.26/abi-spec.html#json) indicates a component `name` is required. The `ABIComponent` type is correct but this function just returns the `type` for nameless components of type `Dict[str, Any]`.